src/Tribe/Asset/Script_Data.php

```diff
@@ -0,0 +1,56 @@
+<?php
+/**
+ * Handles adding script data to the page in cases where localizing a
+ * specific script is not suitable.
+ *
+ * Should generally be accessed via tribe( 'tribe.asset.script-data' )
+ * rather than via direct instantiation.
+ */
+class Tribe__Asset__Script_Data {
+	/**
+	 * Container for any JS data objects that should be added to the page.
+	 *
+	 * @var array
+	 */
+	protected $objects = array();
+
+	/**
+	 * Hooks up the method used to actually render the JSON data.
+	 */
+	public function hook() {
+		if ( is_admin() ) {
+			add_action( 'admin_footer', array( $this, 'render_json' ) );
+		} else {
+			add_action( 'wp_footer', array( $this, 'render_json' ) );
+		}
+	}
+
+	/**
+	 * Adds the provided data to the list of objects that should be available
+	 * to other scripts.
+	 *
+	 * @param string $object_name
+	 * @param array  $data
+	 */
+	public function add( $object_name, array $data ) {
+		$this->objects[ $object_name ] = $data;
+	}
+
+	/**
+	 * Outputs the
+	 * @internal
+	 */
+	public function render_json() {
+		if ( empty( $this->objects ) ) {
+			return;
+		}
+
+		echo '<script type=\'text/javascript\'> /* <![CDATA[ */';
+
+		foreach ( $this->objects as $object_name => $data ) {
+			echo "var $object_name = " . wp_json_encode( $data ) . ';';

```

Comment:
We can - I was actually taking my lead here from how WP does things (the name of the data to localize is not escaped, which makes sense since its entirely under our control) but extra safety isn't a bad thing